### PR TITLE
add rrelatime mount test

### PR DESCRIPTION
**Pull request by lengrongfu**
issue: https://github.com/containers/youki/issues/1534
- [x] rrelatime
- [x] rnorelatime
- [x] rnoatime
- [x] rstrictatime

### CODE REVIEW

**Comment by lengrongfu** [comment]
> Can you run `make lint` to make sure clippy passes? Or we can wait to merge this after we fix the CI.

Let me see.

**Comment by yihuaf** [comment]
Can you rebase to main so we can test this PR with CI now that #1643 is merged :)

**Comment by lengrongfu** [comment]
> Can you rebase to main so we can test this PR with CI now that #1643 is merged :)

Ok, already rebase main.

**Comment by yihuaf** [comment]
Apologize, we may need to merge #1638 first so the CI can be fixed.

**Comment by yihuaf** [comment]
#1638 merged. Please rebase and I can review.

File: tests/rust-integration-tests/integration_test/src/tests/mounts_recursive/mod.rs [hunk 1]
```diff
@@ -397,6 +397,101 @@ fn check_recursive_readwrite() -> TestResult {
     result
 }
 
+fn check_recursive_rrelatime() -> TestResult {
+    let rrelatime_base_dir = PathBuf::from_str("/tmp").unwrap();
+    let rrelatime_dir_path = rrelatime_base_dir.join("rrelatime_dir");
+    let rrelatime_suddirr_path = rrelatime_dir_path.join("rrelatime_subdir");
```
Comment (yihuaf) [inline tests/rust-integration-tests/integration_test/src/tests/mounts_recursive/mod.rs:403]:
```suggestion
    let rrelatime_suddir_path = rrelatime_dir_path.join("rrelatime_subdir");
```

**Review by yihuaf — DISMISSED** [review]

**Comment by yihuaf** [comment]
The tests LGTM. Pending explaination on the need to change `.gitignore`.

**Comment by yihuaf** [comment]
Also, you don't have to do it in the PR, but can you follow up to add more comments explain how the mod test work? I had to trace the code myself to understand where the condition is checked. A simple explanation at top explaining that `runtimetest` will check if the actual mount matches the spec will be good. Otherwise, the reader's first reaction is: I see the spec contains the mount options, but where is the options actually checked.

File: .gitignore [hunk 2]
```diff
@@ -20,3 +20,5 @@ tags.temp
 
 /bundle.tar.gz
 /test.log
+/tests/integration/
```
[the remark this answers is not in the document]
Comment (yihuaf) [reply .gitignore:23]: What about `/contrib`?

[the remark this answers is not in the document]
Comment (yihuaf) [reply .gitignore:23]: > I added a comprehensive test directory to solve this issue

Is the changes for adding this directory already merged or this is prepare ahead?

**Review by yihuaf — APPROVED** [review]

**Pull request merged**